prime-router/src/main/kotlin/azure/ActionHistory.kt

```diff
@@ -595,6 +601,24 @@ class ActionHistory(
         reportFile.blobDigest = blobInfo.digest
         reportFile.bodyUrl = blobInfo.blobUrl
 
+        reportEventService.createItemEvent(
+            childReport = reportFile,
+            eventName = ReportStreamEventName.REPORT_SENT,
+            pipelineStepName = TaskAction.send
+        ) {
+            parentReportId(header.reportFile.reportId)
+            params(
+                mapOf(
+                    ReportStreamEventProperties.TRANSPORT_TYPE to transportType,
+                    ReportStreamEventProperties.RECEIVER_NAME to receiver.fullName
+                ) + if (filename != null) {
```

Comment:
A nice pattern for this kind of thing is using the library function `listOfNotNull` and then calling `.toMap`. Logically equivalent to what you already have but makes adding additional nullable params in the future more simple.

```kotlin
listOfNotNull(  
  ReportStreamEventProperties.TRANSPORT_TYPE to transportType,
  ReportStreamEventProperties.RECEIVER_NAME to receiver.fullName,
  filename?.let { ReportStreamEventProperties.FILENAME to it }
).toMap
```